### PR TITLE
refactor : simplestaking is DAOContract

### PR DESCRIPTION
**Pull request by aliberkhsl**
#99

### CODE REVIEW

File: contracts/staking/BaseShareField.sol [hunk 1]
```diff
@@ -43,29 +26,26 @@ contract BaseShareField is DAOContract, DSMath {
...
-		require(
-			msg.sender == nameService.getAddress("FUND_MANAGER"),
-			"Only FundManager can call this method"
-		);
-		_;
-	}
+	/**
+	 * @dev Helper function to check if caller is fund manager
+	 */
+	function _onlyFundManager() internal virtual {}
```
Comment (sirpy) [inline contracts/staking/BaseShareField.sol:34]: _cantMintRewards

File: contracts/staking/SimpleStaking.sol [hunk 2]
```diff
@@ -441,4 +451,12 @@ contract SimpleStaking is AbstractGoodStaking, StakingToken {
 		(, int256 tokenPriceinUSD, , , ) = tokenPriceOracle.latestRoundData();
 		return (uint256(tokenPriceinUSD) * _amount) / (10**token.decimals()); // tokenPriceinUSD in 8 decimals and _amount is in Token's decimals so we divide it to Token's decimal at the end to reduce 8 decimals back
 	}
+
+	function _onlyFundManager() internal override{
+		require(
+			msg.sender == nameService.getAddress("FUND_MANAGER"),
```
Comment (sirpy) [inline contracts/staking/SimpleStaking.sol:457]: addresses[nameServce.FUND_MANAGER()]

File: contracts/staking/StakingToken.sol [hunk 3]
```diff
@@ -51,9 +51,6 @@ contract StakingToken is BaseShareField {
 			// burn
 			totalSupply -= value;
 		}
-
-		_decreaseProductivity(from, value);
-		_increaseProductivity(to, value);
```
Comment (sirpy) [inline contracts/staking/StakingToken.sol:56]: so now i dont think you need to overwrite _transfer at all

Comment (sirpy) [reply contracts/staking/StakingToken.sol:56]: but where does that happen now if not here?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: contracts/staking/StakingToken.sol (hunk 3 above)
Comment (aliberkhsl) [reply contracts/staking/StakingToken.sol:56]: It's in the simplestaking _transfer override

File: contracts/staking/SimpleStaking.sol (hunk 2 above)
Comment (aliberkhsl) [reply contracts/staking/SimpleStaking.sol]: changed

**Pull request merged**